src/java/org/apache/sqoop/mapreduce/RawKeyTextOutputFormat.java

```diff
@@ -61,6 +62,9 @@ private void writeObject(Object o) throws IOException {
       if (o instanceof Text) {
         Text to = (Text) o;
         out.write(to.getBytes(), 0, to.getLength());
+      } else if (o instanceof BytesWritable) {
```

Comment:
I think it would be a good idea to keep the BinaryKeyOutputFormat because we change the behavior of RawKeyTextOutputFormat now which may lead to issues in other connectors and its name suggests that it handles text and not binary. You can use a technique to elimiate the code duplication similar I used in MainframeDatasetBinaryImportMapper.

Reply:
Thanks @szvasas . I need a bit of help with applying the technique on the OutputFormats as they contain inner classes.